src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/entity/ProcurementEvent.java

```diff
@@ -45,4 +53,29 @@
   @Column(name = "updated_at")
   Instant updatedAt;
 
+  /**
+   * Builds an instance from basic details
+   *
+   * @param jaggaerEventId The rfx reference code
+   * @param principal
+   * @return a procurement event
+   */
+  public static ProcurementEvent of(Integer projectId, String eventName, String jaggaerEventId,
+      String ocdsAuthority, String ocidPrefix, String principal) {
+    var procurementEvent = new ProcurementEvent();
+    procurementEvent.setProjectId(projectId);
+    procurementEvent.setEventName(eventName);
+    procurementEvent.setOcdsAuthorityName(ocdsAuthority);
+    procurementEvent.setOcidprefix(ocidPrefix);
+    procurementEvent.setJaggaerEventId(jaggaerEventId);
+    procurementEvent.setCreatedBy(principal); // Or Jaggaer user ID?
+    procurementEvent.setCreatedAt(Instant.now());
+    procurementEvent.setUpdatedBy(principal); // Or Jaggaer user ID?
+    procurementEvent.setUpdatedAt(Instant.now());
+    return procurementEvent;
+  }
+
+  public String getEventID() {
+    return ocidprefix + "-" + ocdsAuthorityName + "-" + id;
```

Comment:
Swap round - should be `authority-prefix-id` (I think - https://standard.open-contracting.org/latest/en/schema/identifiers/ 🤓

Reply:
Nice one - swapping now 👍